dspace-api/src/main/java/org/dspace/util/SolrImportExport.java

```diff
@@ -0,0 +1,454 @@
+				                 "The absolute path for the directory to use for import or export. If none is given, [dspace]/solr-export is used.");
+
+		try
+		{
+			CommandLine line = parser.parse(options, args);
+			if (line.hasOption(HELP_OPTION))
+			{
+				printHelpAndExit(options, 0);
+			}
+
+			if (!line.hasOption(INDEX_NAME_OPTION))
+			{
+				System.err.println("This command requires the index-name option but none was present.");
+				printHelpAndExit(options, 1);
+			}
+			String[] indexNames = line.getOptionValues(INDEX_NAME_OPTION);
+
+			String action = line.getOptionValue(ACTION_OPTION, "export");
+			if ("import".equals(action))
+			{
+				for (String indexName : indexNames)
+				{
+					try
+					{
+						String fromDir = makeDirectoryName(line.getOptionValue(DIRECTORY_OPTION));
+						String solrUrl = makeSolrUrl(indexName);
+						importIndex(indexName, fromDir, line.hasOption(CLEAR_OPTION), solrUrl);
+					}
+					catch (IOException | SolrServerException | SolrImportExportException e)
+					{
+						System.err.println("Problem encountered while trying to import index " + indexName + ".");
+						e.printStackTrace(System.err);
+					}
+				}
+			}
+			else if ("export".equals(action))
+			{
+				for (String indexName : indexNames)
+				{
+					try
+					{
+						String toDir = makeDirectoryName(line.getOptionValue(DIRECTORY_OPTION));
+						String solrUrl = makeSolrUrl(indexName);
+						String timeField = makeTimeField(indexName);
+						exportIndex(indexName, toDir, solrUrl, timeField);
+					}
+					catch (SolrServerException | IOException | SolrImportExportException e)
+					{
+						System.err.println("Problem encountered while trying to export index " + indexName + ".");
+						e.printStackTrace(System.err);
+					}
+				}
+			}
+			else if ("reindex".equals(action))
+			{
+				for (String indexName : indexNames)
+				{
+					String tempIndexName = indexName + "-temp";
+
+					String origSolrUrl = makeSolrUrl(indexName);
+					String baseSolrUrl = StringUtils.substringBeforeLast(origSolrUrl, "/"); // need to get non-core solr URL
+					String tempSolrUrl = baseSolrUrl + "/" + tempIndexName;
+
+					String solrInstanceDir = ConfigurationManager.getProperty("dspace.dir") + File.separator + "solr" + File.separator + indexName;
+
+					String exportDir = makeDirectoryName(line.getOptionValue(DIRECTORY_OPTION)); // TODO optionally, keep the export?
+					String timeField = makeTimeField(indexName);
+
+					// Create a temp directory to store temporary core data
+					File tempDataDir = new File(ConfigurationManager.getProperty("dspace.dir") + File.separator + "temp" + File.separator + "solr-data");
+					tempDataDir.mkdirs();
+
+					// create a temporary core to hold documents coming in during the reindex
+					try {
+						HttpSolrServer adminSolr = new HttpSolrServer(baseSolrUrl);
+						CoreAdminRequest.Create createRequest = new CoreAdminRequest.Create();
+						createRequest.setInstanceDir(solrInstanceDir);
```

Comment:
Hi @helix84 , the core swapping should actually work with remote cores as well, as long as the [dspace]/solr/conf directory is in place on the local machine. It will just create the temporary directory on the local machine. But you're right, the code should at least check that the [dspace]/solr/conf directory is there and contains the two files needed for the swapping, schema.xml and solrconfig.xml.

Or are there any other spots where you're worried about a remote solr server with regards to the core swapping?